ui-tests/src/test/java/io/syndesis/qe/pages/app/AppPage.java

```diff
@@ -0,0 +1,117 @@
+
+import org.apache.log4j.Logger;
+
+import org.openqa.selenium.By;
+
+import com.codeborne.selenide.CollectionCondition;
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.ElementsCollection;
+import com.codeborne.selenide.SelenideElement;
+
+import java.util.Map;
+
+import io.syndesis.qe.pages.interfaces.SyndesisComponent;
+
+/**
+ * Created by sveres on 11/8/17.
+ */
+public class AppPage implements SyndesisComponent {
+
+	private static final Logger log = Logger.getLogger(AppPage.class);
+
+	private static final class Element {
+		public static final SelenideElement ROOT = $(By.cssSelector("syndesis-root"));
+	}
+
+	@Override
+	public SelenideElement rootElement() {
+		return Element.ROOT.shouldBe(Condition.exist);
+	}
+
+	private static String baseurl = "/";
+
+	public SelenideElement getButton(String buttonTitle) {
+		log.info(String.format("searching for button %s", buttonTitle));
+		return this.rootElement().find(By.xpath(String.format("//button[contains(text(), '%s')]", buttonTitle)));
+	}
+
+	//IS THIS METHOD NECESSARY? - YES IT IS!! KEEP IT.
+	public void clickOnFirstVisibleButton(String buttonTitle) {
+
+		log.info(String.format("searching for first visible button %s", buttonTitle));
+		ElementsCollection buttonElements = this.rootElement().findAll(By.linkText(buttonTitle)).filter(Condition.visible);
+		buttonElements.get(0).click();
+
+	}
+
+	public void clickButton(String buttonTitle) {
+		log.info(String.format("clicking button %s", buttonTitle));
+
+		SelenideElement buttonElement = this.getButton(buttonTitle).shouldBe(Condition.visible);
+
+		buttonElement.click();
+	}
+
+	/**
+	 * Fill form with given data. It will look for ui element for every map entry.
+	 *
+	 * @param data key,value data. Key is used for element lookup.
+	 * @param parrentElement search inputs in child elements of this one
+	 * @param using means what kind of identificator.it is.
+	 * @returns {Promise<[void,T2,T3,T4,T5,T6,T7,T8,T9,T10]>}
+	 */
+	public void fillForm(Map<String, String> data, SelenideElement parrentElement, String using) {
+
+		String value;
+		for (String key : data.keySet()) {
+			value = data.get(key);
+			log.info(String.format("filling form item %s => %s", key, value));
+
+			SelenideElement inputElement = parrentElement.find(By.cssSelector(String.format("input[%s=\"%s\"", using, key))).shouldBe(Condition.visible);
+
+			inputElement.sendKeys(value);
+		}
+	}
+
+	/*
+	* @param select Combobox select element.
+	* @param optionNumber number of the option from list of options.
+	*/
+	public void selectFromDropDownByOptionNumber(SelenideElement select, Integer optionNumber) {
+
+		SelenideElement selectGroup = select.find(By.xpath("'.."));
+
+		SelenideElement dropDownToggle = selectGroup.find(By.className("dropdown-toggle"));
+
+		dropDownToggle.click();
+		dropDownToggle.click();
+
+		ElementsCollection opSelectArray = selectGroup.findAll(By.cssSelector("li")).shouldBe(CollectionCondition.sizeGreaterThanOrEqual(0));
```

Comment:
It would be nice to have all these locator objects statically defined similar way as root locator.